### PR TITLE
build latest released nexus version

### PR DESCRIPTION
**Pull request by joshua-rutherford**
@chrisbsmith, so this will automatically detect the latest version of Nexus, build and publish it via `make` and `make publish`.

Once this is merged we should discuss the cron circle job to publish updates periodically instead of just on merges.  Also, will need to discuss tracking 'latest' in flux.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+MAJOR_PATTERN := ^[0-9]\+
+MINOR_PATTERN := ^[0-9]\+\.[0-9]\+
+PATCH_PATTERN := ^[0-9]\+\.[0-9]\+\.[0-9]\+
+BUILD_PATTERN := ^[0-9]\+\.[0-9]\+\.[0-9]\+\-[0-9]\+
+VERSION_COMMAND := docker run -it deciphernow/nexus:latest cat /usr/local/share/nexus/VERSION
```
Comment (chrisbsmith) [inline Makefile:5]: Does this actually get the latest version or will it catch the N-1 version of the image?  Doesn't `make` execute these at launch and not when the call on line 11 is executed?

**Comment by chrisbsmith** [comment]
I really like how you cleaned this up.  I'm all for the automation of this.  If we can get Circle to build on a cron schedule that's even better.  

We'll have to update flux to allow for it to be more autonomous and to allow new images to trigger an update.  To fully utilize the automation you're built here we'll have to grant that permission.

File: Makefile (hunk 1 above)
Comment (joshua-rutherford) [reply Makefile:5]: Fair, I'll have to break those out.

**Review by chrisbsmith — APPROVED** [review]
LGTM.

File: Makefile (hunk 1 above)
Comment (joshua-rutherford) [reply Makefile]: Okay, to overcome make's desire to execute everything up front I've:

1. Made `build` only tag the `latest` image.
1. Added a `tag` target that tags the other versions from the `latest` image.
1. Removed the dependencies between the targets as that results in the same errors as before.

So now you have to:

```
make && make tag && make publish
```

But you get what you want.

**Pull request merged**